### ISSUE TITLE
Attempt to access run-time storage error

### ISSUE THREAD

**Issue by pparuzel** [microsoft/vscode-cpptools#8673]
Bug type: Language Service

**Describe the bug**
- OS and Version: MacOS Monterey 12.1 (M1 processor)
- VS Code Version:
  - Version: 1.63.2
  - Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
  - Date: 2021-12-15T09:38:17.605Z (1 mo ago)
  - Electron: 13.5.2
  - Chromium: 91.0.4472.164
  - Node.js: 14.16.0
  - V8: 9.1.269.39-electron.0
  - OS: Darwin arm64 21.2.0
- C/C++ Extension Version: v1.8.0-insiders2
- Other extensions you installed (and if the issue persists after disabling them):
  - CMake v0.0.17

Even though the program compiles fine on Apple Clang, Clang13, and GCC11 - IntelliSense reports an error in Problems Tab.
![image](https://user-images.githubusercontent.com/29118021/149519009-1fdda0cb-1c65-4beb-be0c-977f26032fc5.png)
or if you change `values[j]` to `values[i]`:
![image](https://user-images.githubusercontent.com/29118021/149519139-69a6dcb8-e9d9-4517-8a70-54fa472a7d09.png)

**Minimal example**

Here's the most minimal example I could create to reproduce:
```cpp
#include <array>

template <std::size_t N>
constexpr auto transform(const std::array<float, N>& arr)
{
    std::array<int, N> values;
    for (std::size_t i{0}; const auto& k : arr)
    {
        constexpr auto j = 0;
        values[j] = k;  // change this j to i and you get a different error as well
        ++i;
    }
    return values;
}

int main()
{
    constexpr auto values = transform(std::array{0.f});
}
```

**Expected behavior**

There should be no errors in Problems Tab as this C++20 code is perfectly valid.

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. I've filed a bug on our shared VS parser at https://developercommunity.visualstudio.com/t/C-IntelliSense-gives-incorrect-error-/1637986 .

(Note to team: I didn't file an internal bug, because this issue seems more widely applicable to many VS users).

**Comment by sean-mcmanus** [comment]
@pparuzel This is fixed with 1.22.11 (or 1.23.1 pre-release).
